systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -2128,6 +2137,344 @@ void testKafkaOffsetsReplicationFactorHigherThanReplicas() {
         KafkaResource.kafkaClient().inNamespace(NAMESPACE).delete(kafka);
     }
 
+    @Test
+    void testDynamicConfiguration() {
+        int kafkaReplicas = 2;
+        Map<String, Object> kafkaConfig = new HashMap<>();
+        kafkaConfig.put("offsets.topic.replication.factor", "1");
+        kafkaConfig.put("transaction.state.log.replication.factor", "1");
+        kafkaConfig.put("default.replication.factor", "1");
+        kafkaConfig.put("log.message.format.version", "2.4");
+
+        Map<String, Object> updatedKafkaConfig = new HashMap<>();
+        updatedKafkaConfig.put("offsets.topic.replication.factor", "1");
+        updatedKafkaConfig.put("transaction.state.log.replication.factor", "1");
+        updatedKafkaConfig.put("default.replication.factor", "1");
+        updatedKafkaConfig.put("log.message.format.version", "2.4");
+        updatedKafkaConfig.put("unclean.leader.election.enable", "true");
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, kafkaReplicas, 1)
+                .editSpec()
+                    .editKafka()
+                        .withConfig(kafkaConfig)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        String kafkaConfiguration = kubeClient().getConfigMap(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).getData().get("server.config");
+        assertThat(kafkaConfiguration, containsString("offsets.topic.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("transaction.state.log.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("default.replication.factor=1"));
+
+        String kafkaConfigurationFromPod = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "bin/kafka-configs.sh --bootstrap-server localhost:9092 --entity-type brokers --entity-name 0 --describe").out();
+        assertThat(kafkaConfigurationFromPod, is("Configs for broker 0 are:\n"));
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Updating configuration of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setConfig(updatedKafkaConfig);
+        });
+
+        PodUtils.waitUntilPodsStability(kubeClient().listPodsByPrefixInName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)));
```

Comment:
Why do you using this method? I assume you are using the dynamic configuration for instant change or?  Do we really need it?

Reply:
The change is not instant. It takes ~5 seconds